src/openrct2/ride/TrackDesign.cpp

```diff
@@ -762,7 +762,7 @@ static std::optional<TrackSceneryEntry> TrackDesignPlaceSceneryElementGetEntry(c
             result.Type = obj->GetObjectType();
             result.Index = objectMgr.GetLoadedObjectEntryIndex(obj);
         }
-        else
+        if (obj == nullptr || (!ResearchIsInvented(result.Type, result.Index) && !gCheatsIgnoreResearchStatus))
```

Comment:
The first condition is the exact inverse of the previous, so imo it's nicer to keep this as an else-block:
```suggestion
        else if (!ResearchIsInvented(result.Type, result.Index) && !gCheatsIgnoreResearchStatus)
```

Reply:
That would defeat the point of this PR. Scenery which is in the scenario (`obj != nullptr`) but not yet researched would then never be marked as unavailable since it would always take first if branch.

Your suggestion would work if we were testing an && condition, but this is an ||.

Reply:
You're right, these two ifs aren't mutually exclusive anymore, and I now notice that this is actually using the values from the previous if block in the condition.

Rewriting this to make it more obvious that there are two conditions for the object's availability would be nice:


```cpp
        auto obj = objectMgr.GetLoadedObject(scenery.scenery_object);
        bool objectIsAvailable = obj != nullptr;
        if (objectIsAvailable)
        {
            result.Type = obj->GetObjectType();
            result.Index = objectMgr.GetLoadedObjectEntryIndex(obj);
            objectIsAvailable = ResearchIsInvented(result.Type, result.Index);
        }

        if (objectIsAvailable && !gCheatsIgnoreResearchStatus))
```